packages/app/src/components-styled/image.tsx

```diff
@@ -8,3 +9,26 @@ export const Image = styled.img(
     height: 'auto',
   })
 );
+
+type SanityImageProps = {
+  src: string;
+  srcSet?: string;
+  css?: CssFunctionReturnType;
+};
+
+export function SanityImage(props: SanityImageProps) {
+  const { src, srcSet, ...imageProps } = props;
+
+  if (!srcSet) {
+    return <Image {...props} />;
+  }
+
+  const extension = src.split('.')[1];
```

Comment:
nit; This doesn't seem like a very robust way to determine a file extension. Maybe take the last item instead of assuming there to be one dot / two components, that would already make it more robust.
```suggestion
  const extension = src.split('.').pop();
```

Reply:
You can also get the sha1hash from sanity and manually add the width and webp extension behind it. That's even safer, because this approach will break whenever an image is uploaded with a dot in the filename. But I'll admit, it's a bit more complicated and we know Sanity returns sha1 hashes anyway. There's no real danger of that ever happening.

Another possibility:

`const [filename, extension] = src.split('.')`

Reply:
i opted for the suggestion to get the last item 👍